### PR TITLE
Polkadot v1.1.0 Integration Tests

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

PR over `polkadot-v1.1.0` containing integration-tests changes

### CODE REVIEW

File: runtime/integration-tests/src/generic/mod.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@ mod impls;
 pub mod utils;
 
 // Test cases
-mod cases {
+/*mod cases {
```
Comment (lemunozm) [inline runtime/integration-tests/src/generic/mod.rs:16]: Uncomment this

File: runtime/integration-tests/src/lib.rs [hunk 2]
```diff
@@ -14,10 +14,10 @@
 #![cfg(test)]
 #![allow(unused)]
 
-mod evm;
+//mod evm;
 mod generic;
-mod rewards;
-mod runtime_apis;
+//mod rewards;
+//mod runtime_apis;
```
Comment (lemunozm) [inline runtime/integration-tests/src/lib.rs:20]: Uncomment this

File: runtime/integration-tests/src/generic/mod.rs (hunk 1 above)
Comment (lemunozm) [inline runtime/integration-tests/src/generic/mod.rs:16]: Uncomment this

File: runtime/integration-tests/src/lib.rs (hunk 2 above)
Comment (lemunozm) [inline runtime/integration-tests/src/lib.rs:20]: Uncomment this

File: runtime/integration-tests/src/utils/accounts.rs [hunk 3]
```diff
@@ -14,24 +14,25 @@
...
-use node_primitives::{AccountId as RelayAccountId, Index as RelayIndex};
+use polkadot_core_primitives::{AccountId as RelayAccountId, Nonce as RelayNonce};
 pub use sp_core::sr25519;
 use sp_core::{
 	sr25519::{Pair, Public, Signature},
 	Pair as PairT,
 };
 use sp_runtime::AccountId32;
 
+/*
```
Comment (lemunozm) [inline runtime/integration-tests/src/utils/accounts.rs:27]: Uncomment this

File: runtime/integration-tests/src/generic/utils/democracy.rs [hunk 4]
```diff
@@ -58,6 +55,7 @@ pub fn fast_track<T: Runtime>(
 	.into()
 }
 
+/*
```
Comment (lemunozm) [inline runtime/integration-tests/src/generic/utils/democracy.rs:58]: Uncomment

File: runtime/integration-tests/src/generic/mod.rs [hunk 5]
```diff
@@ -5,15 +5,15 @@
 
 pub mod env;
 pub mod envs {
-	pub mod fudge_env;
```
Comment (lemunozm) [inline runtime/integration-tests/src/generic/mod.rs:8]: Uncomment

**Comment by lemunozm** [comment]
Fudge updated, now using same `polkadot-sdk` dependencies: [bd15cc8](https://github.com/centrifuge/centrifuge-chain/pull/1776/commits/bd15cc851ee6255f8c0226cf37ea79f43c2fc163)

**Comment by lemunozm** [comment]
Currently failing just 3 test. 
```
---- generic::cases::liquidity_pools::altair::transfers::transfer_ksm_to_and_from_relay_chain::altair stdout ----
thread 'generic::cases::liquidity_pools::altair::transfers::transfer_ksm_to_and_from_relay_chain::altair' panicked at runtime/integration-tests/src/generic/cases/liquidity_pools.rs:5779:17:
assertion `left == right` failed
  left: 1991963000000
 right: 1993570400000
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- generic::cases::liquidity_pools::centrifuge::transfers::transfer_dot_to_and_from_relay_chain::centrifuge stdout ----
thread 'generic::cases::liquidity_pools::centrifuge::transfers::transfer_dot_to_and_from_relay_chain::centrifuge' panicked at runtime/integration-tests/src/generic/cases/liquidity_pools.rs:7900:17:
assertion `left == right` failed
  left: 29919630000
 right: 29935704000

---- generic::cases::liquidity_pools::centrifuge::restricted_transfers::restrict_dot_xcm_transfer::centrifuge stdout ----
thread 'generic::cases::liquidity_pools::centrifuge::restricted_transfers::restrict_dot_xcm_transfer::centrifuge' panicked at runtime/integration-tests/src/generic/cases/liquidity_pools.rs:7530:17:
assertion `left == right` failed
  left: 29919630000
 right: 29935704000


failures:
    generic::cases::liquidity_pools::altair::transfers::transfer_ksm_to_and_from_relay_chain::altair
    generic::cases::liquidity_pools::centrifuge::restricted_transfers::restrict_dot_xcm_transfer::centrifuge
    generic::cases::liquidity_pools::centrifuge::transfers::transfer_dot_to_and_from_relay_chain::centrifuge

test result: FAILED. 217 passed; 3 failed; 0 ignored; 0 measured; 0 filtered out; finished in 9423.86s

error: test failed, to rerun pass `-p runtime-integration-tests --lib`

```

All have as common denominator the call of `fee()`, any idea of what could happen @cdamian?

**Comment by cdamian** [comment]
@lemunozm yes, it seems that the fee calculations that we have there are no longer valid, I manually updated those expected values in the branch that I had here.

**Comment by lemunozm** [comment]
The value of `WEIGHT_REF_TIME_PER_NANOS` has changed from `v0.9.43` to `v1.1.0`. It seems like something that changes any new polkadot version. 

I'm not sure why we do that check. But IMO, we could just remove those lines because they're not related to the test itself. WDYT @cdamian @wischli?

For reference the failing asserts are: 

```rust
assert_eq!(
    orml_tokens::Pallet::<T>::free_balance(currency_id, &Keyring::Bob.into()),
    transfer_amount - fee(meta.decimals)
);
```
```rust
assert_eq!(alice_initial_dot, dot(3) - dot_fee());
```

**Comment by cdamian** [comment]
> The value of `WEIGHT_REF_TIME_PER_NANOS` has changed from `v0.9.43` to `v1.1.0`. It seems like something that changes any new polkadot version.
> 
> I'm not sure why we do that check. But IMO, we could just remove those lines because they're not related to the test itself. WDYT @cdamian @wischli?
> 
> For reference the failing asserts are:
> 
> ```rust
> assert_eq!(
>     orml_tokens::Pallet::<T>::free_balance(currency_id, &Keyring::Bob.into()),
>     transfer_amount - fee(meta.decimals)
> );
> ```
> 
> ```rust
> assert_eq!(alice_initial_dot, dot(3) - dot_fee());
> ```

I wouldn't mind having those checks in there to be honest. I think we might catch any unexpected changes to the fees that are charged when doing those kind of transfers.

**Comment by lemunozm** [comment]
And how would you solve the current issue then? Without hardcoding the values. Or maybe we're having the issue right now, because from what I can see, if it worked before, it should work now

**Comment by cdamian** [comment]
@lemunozm I hardcoded those values in some of the other tests. We can either do that or fix the `fee` funcs if possible.

**Comment by lemunozm** [comment]
The problem with hardcoding is that we're losing the "why it fails". Other would see this failing without any reason on the why. I'll give a try to see why `fee` is failing when it shouldn't

**Comment by lemunozm** [comment]
I'll merge this by now. Issues should be addressed in https://github.com/centrifuge/centrifuge-chain/pull/1756

**Pull request merged**